src/vm/rejit.inl

```diff
@@ -26,7 +26,10 @@ inline BOOL ReJitManager::IsReJITEnabled()
 {
     LIMITED_METHOD_CONTRACT;
 
-    return CORProfilerEnableRejit();
+    static bool profilerStartupRejit = CORProfilerEnableRejit() != FALSE;
+    static ConfigDWORD rejitOnAttachEnabled;
+
+    return  profilerStartupRejit || (rejitOnAttachEnabled.val(CLRConfig::EXTERNAL_ProfAPI_RejitOnAttach) != 0);
```

Comment:
@noahfalk But will that mean that by default .NET Core apps will not get ReJit on Attach? That's a very useful thing to do.

Reply:
>But will that mean that by default .NET Core apps will not get ReJit on Attach? 

Nope : ) The default value of CLRConfig::EXTERNAL_ProfAPI_RejitOnAttach = 1 so by default this API returns true.

Reply:
@noahfalk I thought for some reason there was already a check for enabling rejit separately from the immutable flags, but looking back over the code I don't know where I got that idea from.

I just added a commit to prevent rejit flags from being set unless it's either during startup or is already enabled, and to prevent detach if rejit is enabled. SetEventMask will return `CORPROF_E_REJIT_NOT_ENABLED`, that was the closest error message I could find. Should I add a more specific one since this one was originally intended to mean that you called a rejit api without enabling it, or is overloading the error message ok?

Reply:
Seems ok to me